src/App.tsx

```diff
@@ -15,71 +15,103 @@ export const goods = [
+    setSelectedGood('');
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  const buttonClickHandler = (
+    e: React.MouseEvent<HTMLButtonElement>,
+    good: string,
+  ) => {
+    if (e.currentTarget.textContent === '-') {
+      setSelectedGood('');
+    } else {
+      setSelectedGood(good);
+    }
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
+  return (
+    <main className="section container">
+      {!selectedGood && <h1 className="title">No goods selected</h1>}
+
+      {selectedGood && (
+        <h1 className="title is-flex is-align-items-center">
+          {`${selectedGood} is selected`}
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          {selectedGood && (
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={clearHandler}
+              aria-label="Clear selection"
+            />
+          )}
+        </h1>
+      )}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+      <table className="table">
+        <tbody>
+          {goods.map((good) => {
+            let tableRow;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+            if ((selectedGood && selectedGood !== good) || !selectedGood) {
+              tableRow = (
+                <tr data-cy="Good" key={good}>
+                  <td>
+                    <button
+                      data-cy="AddButton"
+                      type="button"
+                      className="button"
+                      onClick={(e) => {
+                        buttonClickHandler(e, good);
+                      }}
+                    >
+                      +
+                    </button>
+                  </td>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+                  <td data-cy="GoodTitle" className="is-vcentered">
+                    {good}
+                  </td>
+                </tr>
+              );
+            }
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+            if (selectedGood === good) {
```

Comment:
can we use here just `else`?

Reply:
We can use just return here
`return (
              <tr
                className="has-background-success-light"
                data-cy="Good"
                key={good}
              >`